### ISSUE TITLE
[firebase_auth] Getting Session expired error

### ISSUE THREAD

**Issue by Varun-BP** [FirebaseExtended/flutterfire#217]
```
I am getting session expired error like below,

E/flutter (23252): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: PlatformException(ERROR_SESSION_EXPIRED, The sms code has expired. Please re-send the verification code to try again., null)
E/flutter (23252): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:564:7)
E/flutter (23252): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:316:33)
E/flutter (23252): <asynchronous suspension>
E/flutter (23252): #2      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:344:48)
E/flutter (23252): <asynchronous suspension>
E/flutter (23252): #3      FirebaseAuth.signInWithCredential (package:firebase_auth/src/firebase_auth.dart:257:23)
E/flutter (23252): <asynchronous suspension>
E/flutter (23252): #4      _GetOtpState._signInWithPhoneNumber (package:goritzy/test2.dart:94:22)
E/flutter (23252): <asynchronous suspension>
E/flutter (23252): #5      _GetOtpState._submit (package:goritzy/test2.dart:53:7)
```

**Comment by iapicca** [comment]
Hi @Varun-BP 
could you please provide your `flutter doctor -v`
and your `flutter run --verbose`.
Also, to better address the issue, would be helpful
if you could post a self contained app on github
or the steps to reproduce it.
Thank you

**Comment by phong764119** [comment]
I've got the same issue, i used the example code from github and when i tried using sms Phone Authentication and added my phone number for testing(fixed verification code) it's worked. But when i tried add SHA-1 key and re-login. it did send verification code but i've got the error ERROR_SESSION_EXPIRED, i check  firebase console it shows my phone number is registried, so i guess that the android is auto add verification code and authorized. 
Btw the same code is working well in IOS

**Comment by iapicca** [comment]
Hi @phong764119 
could you please provide your flutter doctor -v
and your flutter run --verbose.
Also, to better address the issue, would be helpful
if you could post a self contained app on github
or the steps to reproduce it.
Thank you

**Comment by phong764119** [comment]
Hi @iapicca 
Thanks you for response, 
The code i used is exactly from [Firebase example](https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_auth/firebase_auth/example), it worked as i expected.
If i added my phone number to Phone numbers for testing in firebase console and the the default code is 123456 it's worked everytime when i try login and re-login
But when i tried to use real device and phone number (the sim card on the same device) the firebase server will send a code verification and it'll automatically certified without adding verify code and trigger event [PhoneVerificationCompleted](https://github.com/FirebaseExtended/flutterfire/blob/66307dae6d8b3a0b5f452f26d3e9abd7fde05724/packages/firebase_auth/firebase_auth/example/lib/signin_page.dart#L519 )
So basically it still work, but is there any way to get verify code and auto add to the input (for user UI experience) so he/she can noted that the code is auto get from SMS 
Secondly, when i logout and re-login , the firebase doesn't send any SMS but the app still can login anyway

**Comment by Zubi1001** [comment]
> Hi @iapicca
> Thanks you for response,
> The code i used is exactly from [Firebase example](https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_auth/firebase_auth/example), it worked as i expected.
> If i added my phone number to Phone numbers for testing in firebase console and the the default code is 123456 it's worked everytime when i try login and re-login
> But when i tried to use real device and phone number (the sim card on the same device) the firebase server will send a code verification and it'll automatically certified without adding verify code and trigger event [PhoneVerificationCompleted](https://github.com/FirebaseExtended/flutterfire/blob/66307dae6d8b3a0b5f452f26d3e9abd7fde05724/packages/firebase_auth/firebase_auth/example/lib/signin_page.dart#L519)
> So basically it still work, but is there any way to get verify code and auto add to the input (for user UI experience) so he/she can noted that the code is auto get from SMS

actually. if the user has google play services running on his device, he won't really receive an otp and his device will auto register.

**Comment by Ehesp** [comment]
Hi @Varun-BP please could you confirm this is still an issue on the latest Flutter & auth plugin version please? Thanks.

**Comment by kaumudpa** [comment]
I am facing the same problem on android device.

**Comment by Varun-BP** [comment]
> Hi @Varun-BP please could you confirm this is still an issue on the latest Flutter & auth plugin version please? Thanks.

No, now i am not facing this problem

**Comment by Allviss0211** [comment]
Has anyone tried to modify the expiration time of the otp code?
